crawl-ref/source/godabil.cc

```diff
@@ -3792,6 +3793,32 @@ bool ashenzari_end_transfer(bool finished, bool force)
     return true;
 }
 
+bool ashenzari_curse_item()
+{
+    while (1)
+    {
+        int item_slot = prompt_invent_item("Curse which item?", MT_INVLIST,
+                                           OSEL_CURSABLE,
+                                           true, true, false);
+        if (prompt_failed(item_slot))
+            return false;
+
+        item_def& item(you.inv[item_slot]);
+
+        ASSERT(!item.cursed());
+        if (!item_is_equipped(item))
```

Comment:
Change the OSEL to only be equipped uncursed items and change this line to an ASSERT